.circleci/config.yml

```diff
@@ -218,36 +237,121 @@ jobs:
+          - ui-cache-{{ checksum "~/workbench/ui/package.json" }}
+          - ui-cache-
+      - run:
+          working_directory: ~/workbench/ui
+          command: yarn install
+      - save_cache:
+          paths:
+            - ~/workbench/ui/node_modules
+          key: ui-cache-{{ checksum "~/workbench/ui/package.json" }}
       - deploy:
-          name: Maybe deploy to test/staging
+          name: Deploy to App Engine
           working_directory: ~/workbench/ui
           command: |
             if [ "${CIRCLE_BRANCH}" == "master" ]; then
               ../ci/activate_creds.sh circle-sa-key.json
               ./project.rb deploy-ui \
                 --project all-of-us-workbench-test \
-                --account=circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
+                --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
                 --version circle-ci-test \
                 --promote
             elif [ "${CIRCLE_TAG}" == "v*" ]; then
               ../ci/activate_creds.sh circle-sa-key.json
               ./project.rb deploy-ui \
                 --project all-of-us-rw-staging \
-                --account=circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
+                --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
                 --version "${CIRCLE_TAG}" \
                 --promote
             else
-              echo "Not master branch or tagged release, skipping deploy; building only"
-              ./project.rb build --environment test
+              echo "Not master branch or tagged release, skipping deploy"
             fi
 
 workflows:
   version: 2
   build-test-deploy:
     jobs:
-      - api-build-test-deploy
-      - api-local-tests
-      - ui-build-test-deploy
+      # Always run basic test/lint/compilation (open PRs, master merge, tags).
+      - api-local-test:
+          filters:
+            tags:
+              only: /^v.*/
+      - api-build-test:
+          filters:
+            tags:
+              only: /^v.*/
+      - ui-build-test:
+          filters:
+            tags:
+              only: /^v.*/
+      # Run slower integration tests and dep checks on release tag.
+      - api-bigquery-test:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+      - api-deps-check:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+      - api-integration-test:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+      # Run deployment to "test" on master merge, or "staging" on release tag.
+      - api-deploy:
+          requires:
+            - api-local-test
+            - api-build-test
+            - api-bigquery-test
+            - api-deps-check
+            - api-integration-test
```

Comment:
We didn't run integration or bigquery tests before on every master merge... I'm not sure we should start now. Can we do these only for a release to staging?

Reply:
Barring any bugs, that is what this is already doing. Added comment.